services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/Consumer.java

```diff
@@ -47,32 +73,38 @@ public void run() {
              *  - max.poll.interval.ms = 300000 (ie 300 seconds)
              *  - max.poll.records = 500 (must be able to process about 2 records per second
              */
-            KafkaConsumer<String, String> consumer = null;
-            try {
-                consumer = new KafkaConsumer<>(context.getKafkaConsumerProperties());
+            try (KafkaConsumer<String, String> consumer =
+                         new KafkaConsumer<>(kafkaConfig.createKafkaConsumerProperties())) {
                 consumer.subscribe(topics);
 
+                KafkaOffsetRegistry offsetRegistry =
+                        new KafkaOffsetRegistry(consumer, kafkaConfig.getAutoCommitInterval());
+
                 while (true) {
                     ConsumerRecords<String, String> batch = consumer.poll(100);
-                    if (batch.count() < 1) {
+                    if (batch.isEmpty()) {
                         continue;
                     }
 
                     logger.debug("Received records batch contain {} messages", batch.count());
+
                     for (ConsumerRecord<String, String> record : batch) {
                         handle(record);
+
+                        offsetRegistry.addAndCommit(record);
                     }
+
+                    // force to commit after each completed batch.
+                    offsetRegistry.commitOffsets();
+
                     switchManager.safeModeTick(); // HACK alert .. should go in its own timer loop
                 }
+            } catch (InterruptException ex) {
+                // Leave if the thread has been interrupted.
+                throw ex;
```

Comment:
We leave in uncommitted state some set of processed messages. So each FL shutdown will(can) create a gap in commited/processed messages and in worst case lead to repeated processing of this messages.

Reply:
Good point. I'll try to address this.

Reply:
Fixed.